### PR TITLE
fix: modify the priority order of `source` field and the webpack default mainFields

### PR DESCRIPTION
**Pull request by targeral**
### <samp>🤖 Generated by Copilot at 083483a</samp>

This pull request fixes the priority order of the `source` field in the `@modern-js/builder` package. It updates the `sourceBuild.ts` file and the changelog for a patch release.

### <samp>🤖 Generated by Copilot at 083483a</samp>

* Fix the priority order of the `source` field and the webpack default mainFields in `@modern-js/builder` package ([link](https://github.com/web-infra-dev/modern.js/pull/4889/files?diff=unified&w=0#diff-d398fabb54647edd1f8cb54f3d0db470898056b22f1b780ea4cac0e9832fb2f3L111-R113), [link](https://github.com/web-infra-dev/modern.js/pull/4889/files?diff=unified&w=0#diff-4d37ba101c41911c167aedaa64d3fb2609a90319ff69d54af81a8f934cca9534R1-R6))
  - Put the `sourceField` first in the `mainFields` array in the `builderPluginSourceBuild` function in `sourceBuild.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4889/files?diff=unified&w=0#diff-d398fabb54647edd1f8cb54f3d0db470898056b22f1b780ea4cac0e9832fb2f3L111-R113))
  - Add a comment to explain why the `resolve.conditionNames` property is not used ([link](https://github.com/web-infra-dev/modern.js/pull/4889/files?diff=unified&w=0#diff-d398fabb54647edd1f8cb54f3d0db470898056b22f1b780ea4cac0e9832fb2f3L111-R113))
  - Update the changelog with a patch version and a summary of the fix in both English and Chinese ([link](https://github.com/web-infra-dev/modern.js/pull/4889/files?diff=unified&w=0#diff-4d37ba101c41911c167aedaa64d3fb2609a90319ff69d54af81a8f934cca9534R1-R6))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder/src/plugins/sourceBuild.ts [hunk 1]
```diff
@@ -108,7 +108,9 @@ export function builderPluginSourceBuild(
             // webpack.js.org/configuration/module/#ruleresolve
             chain.module
               .rule(useTsLoader ? CHAIN_ID.RULE.TS : CHAIN_ID.RULE.JS)
-              .resolve.mainFields.merge(['...', sourceField]);
+              // source > webpack default mainFiedls.
+              // when source is not exist, other mainFields will effect.
+              .resolve.mainFields.merge([sourceField, '...']);
```
Comment (chenjiahan) [inline packages/builder/builder/src/plugins/sourceBuild.ts:113]: Should we update the `conditionNames` orders too?

Comment (targeral) [reply packages/builder/builder/src/plugins/sourceBuild.ts:113]: No, `conditionNames` is different from `mainFields`. During condition matching, earlier entries have higher priority and take precedence over later entries  in the `exports` object.

Comment (chenjiahan) [reply packages/builder/builder/src/plugins/sourceBuild.ts:113]: get👌

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Please submit a PR to rsbuild repo too: https://github.com/web-infra-dev/rsbuild

**Comment by targeral** [comment]
> Please submit a PR to rsbuild repo too: https://github.com/web-infra-dev/rsbuild ❤️

OK

**Pull request merged**